src/mon/OSDMonitor.cc

```diff
@@ -13033,6 +13033,12 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     if (pp->snap_exists(snapname.c_str())) {
       ss << "pool " << poolstr << " snap " << snapname << " already exists";
     } else {
+      if (const auto& fsmap = mon.mdsmon()->get_fsmap(); fsmap.pool_in_use(pool)) {
```

Comment:
From the commit description:

> Pool level snaps and mon managed snaps can't co-exist (...)

Isn't this a thinko? Per https://github.com/ceph/ceph/pull/47753#discussion_r953292185 I started thinking it's about _Pool level snaps and **FS level snaps**_.

Reply:
yes; this is about FS level snaps and Pool level snaps
its just the nomenclature that @gregsfortytwo has used to refer to it (AFAIUI)

but looks like the phrasing should be changed to _FS level snaps and Pool level snaps can't coexist_

Reply:
phrasing changed to _"Pool-level and fs-level snapshots ..._"